src/DotLiquid.Tests/Util/ObjectExtensionMethodsTests.cs

```diff
@@ -1,122 +1,229 @@
+
+            Assert.That(1.SafeTypeInsensitiveEqual("1"), Is.False); // int to string equality
+            Assert.That(2.0f.SafeTypeInsensitiveEqual("2.0"), Is.False);  // float to string equality
+            Assert.That(2.0d.SafeTypeInsensitiveEqual("2.0"), Is.False);  // double to string equality
+
+            Assert.That(true.SafeTypeInsensitiveEqual("true"), Is.False); // bool to Truthy string equality
+            Assert.That("true".SafeTypeInsensitiveEqual(true), Is.False); // Truthy string to bool equality
+            Assert.That("true".SafeTypeInsensitiveEqual("TRUE"), Is.False); // Truthy string to Truthy string equality
+
+            Assert.That(false.SafeTypeInsensitiveEqual("false"), Is.False); // bool to Falsy string equality
+            Assert.That("false".SafeTypeInsensitiveEqual(false), Is.False); // Falsy string to bool equality
+            Assert.That("false".SafeTypeInsensitiveEqual(false), Is.False); // Falsy string to bool equality
+            Assert.That("false".SafeTypeInsensitiveEqual("FALSE"), Is.False); // Falsy string to Falsy string equality
+
+            // Equals
+            Assert.That(NIL.SafeTypeInsensitiveEqual(null), Is.True); // null equalilty
+            Assert.That("a string".SafeTypeInsensitiveEqual("a string"), Is.True); // same type equality
+            Assert.That(Int64.Parse("99").SafeTypeInsensitiveEqual(Int32.Parse("99")), Is.True); // long to int equality
+            Assert.That(true.SafeTypeInsensitiveEqual(true), Is.True); // bool to bool equality
+            Assert.That(false.SafeTypeInsensitiveEqual(false), Is.True); // bool to bool equality
+            Assert.That("A".SafeTypeInsensitiveEqual('A'), Is.True); // string to char equality
+            Assert.That('A'.SafeTypeInsensitiveEqual("A"), Is.True); // char to string equality
+            Assert.That(StringComparison.Ordinal.SafeTypeInsensitiveEqual("Ordinal"), Is.True); // enum to string equality
+            Assert.That("Ordinal".SafeTypeInsensitiveEqual(StringComparison.Ordinal), Is.True); // string to enum equality
+        }
+
+        [Test]
+        public void TestSafeTypeInsensitiveEqual_NumericTypes()
+        {
+            byte valueU8 = 12;
+            ushort valueU16 = 12;
+            uint valueU32 = 12;
+            ulong valueU64 = 12;
+            sbyte value8 = 12;
+            short value16 = 12;
+            int value32 = 12;
+            long value64 = 12;
+            decimal valueDecimal = 12;
+            float valueFloat = 12;
+            double valueDouble = 12;
+            object[] equalValues = {
+                valueU8,
+                valueU16,
+                valueU32,
+                valueU64,
+                value8,
+                value16,
+                value32,
+                value64,
+                valueDecimal,
+                valueFloat,
+                valueDouble
+            };
+            for (int i = 0; i < equalValues.Length; i++)

```

Comment:
```suggestion

            for (int i = 0; i < equalValues.Length; i++)
```

Separating logic from bloc of variables declaration.